apps/project/tests/test_apis.py

```diff
@@ -538,6 +538,31 @@ def test_update_project_remove_ug(self):
             final_member_count
         )
 
+    def test_duplicate_usergroup_add_in_project(self):
+        project = self.create(
+            Project,
+            title='For test',
+            user_groups=[],
+            role=self.admin_role
+        )
+        # add usergroup to the project
+        ProjectUserGroupMembership.objects.create(
+            usergroup=self.ug1,
+            project=project
+        )
+
+        # now try to create same usergroup from api level
+        data = {
+            'usergroup': self.ug1.id,
+            'role': self.normal_role.id
+        }
+        url = f'/api/v1/projects/{project.id}/project-usergroups/'
+        self.authenticate()
+        response = self.client.post(url, data)
+        self.assert_400(response)
```

Comment:
Let's add a retry after removing the membership we added manually.

Reply:
Done!